### DISCUSSION TITLE
How to watch external asset files?

### DISCUSSION THREAD

**Discussion by Drakota** [parcel-bundler/parcel#7722] [Q&A] [answered]
Hi, 
I'm using a URL import to fetch one of my asset and I have an external watcher that might modify it and I'd like to be able to get Parcel to refresh in sync with everything. I have this import:
```ts
import asm from "url:../../target/wasm32-unknown-unknown/debug/source.wasm";
```
I've setup a nodemon watcher that reloads the parcel watch server like this:
```json
    "watch": "nodemon -e wasm --watch ../target -x 'npm run start'",
```
But I'm losing connection to the HMR server and I need to refresh manually. Is there a way to do something like this with a Parcel plugin or a built-in feature?

I know that Parcel 1 add WASM and Rust support for automatically recompiling a project and I know it's a work in progress for Parcel 2, but for my use case all the building is already handled by something else, I just want to watch "some" file for changes and reload the Parcel watch server, hence the title being generic to an asset file.

**Comment by mischnic** [comment, marked as answer]
I would leave Parcel running, and then just recompile the Rust part separate (either manually after changes or using some watcher to run `cargo build`).
After the Rust build has written a new wasm file, Parcel would detect that automatically and reload as appropriate. (So all assets that you reference from JS/HTML are automatically watched)
